crux-test/test/crux/tx_test.clj

```diff
@@ -663,3 +663,17 @@
                (api/q (api/db *api*) '{:find [parent child]
                                        :where [[parent :crux.db/id _]
                                                [child :joins parent]]}))))))
+
+(t/deftest tx-doc-hooks-422
+
+  (t/testing "doc hooks"
+    (let [test-atom-doc (atom false)]
+      (db/add-doc-hook! (:indexer *api*) (fn [_] (fn [_] (swap! test-atom-doc not))))
```

Comment:
why the double fn here, out of interest?

Reply:
oh, I see what's happening here - first function is the pre-hook, returns the function that you want to run after the transaction? interesting, I've not seen hooks implemented this way (and we probably shouldn't be inventing a new standard!)

for middleware, the hook usually accepts the underlying implementation as a parameter, and then has the opportunity to augment it both before and after the call. e.g.:

```clojure
(fn my-hook [f]
  (fn [params]
    ;; pre-hook work here, alter `params` if need be
    (let [ret (f params)]
      ;; post-hook work here, alter `ret` if need be
      ret)))
```